source/IO.cpp

```diff
@@ -368,8 +368,9 @@ IO::Serialize(const quaternion& m)
 {
 	std::vector<uint64_t> data;
 	auto coeffs = m.coeffs();
-	data.reserve(coeffs.size());
-	for (unsigned int i = 0; i < 4; i++)
+	data.reserve(coeffs.size() + 1);
+	data.push_back(Serialize((uint64_t)coeffs.size()));
+	for (unsigned int i = 0; i < coeffs.size(); i++)
```

Comment:
Is there a reason to serialize quaternions to include their size? They should always have 4 coefficients in the same way that a vec3 has 3 coefficients, a mat has 9, etc.

Reply:
Yeah, I was afraid it can have less than 4 components. But apparently it cannot... I am just assuming 4

Reply:
Argh, stupid error fixed on https://github.com/KelsonMarine/MoorDyn_Public/pull/1/commits/c19eb9621d1ad7dd0d79c63ba945f833ca24a94a